beacon-chain/rpc/prysm/v1alpha1/validator/aggregator.go

```diff
@@ -32,10 +33,18 @@ func (vs *Server) SubmitAggregateSelectionProof(ctx context.Context, req *ethpb.
 		return nil, err
 	}
 
-	st, err := vs.HeadFetcher.HeadStateReadOnly(ctx)
+	st, err := vs.HeadFetcher.HeadState(ctx)
```

Comment:
How about just grab the headslot first and check if it's the current slot, so that then we can deal with the read only case and not make a copy? this should be essentially every single case except if we're aggregating on a reorg. I would probably go further and only really check first for justification, if the justified checkpoint is two epochs ago, then the shuffling hasn't changed and we can use headstate read only too.